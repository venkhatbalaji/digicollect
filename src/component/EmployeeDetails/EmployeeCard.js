import React from "react";
import styled from "styled-components";

const Card = styled.div`
    height: 300px;
    width: 250px;
    background: white;
    border-radius: 10px;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    padding: 5px;
    margin: 10px;
    gap: 15px;
`;

const CardTitle = styled.h1`
    font-size: 15px;
    color: ${props => props.color};
`;

const CardImage = styled.img`    
    background-color: white;
    border-radius: 100px;
    height: 100px;
    width: 100px;
    border: 3px solid ${props => props.color};
    border-bottom-color: white;
    border-left-color: white;
`;

const CardDesignation = styled.p`
    background: ${props => props.color};
    border-radius: 5px;
    padding: 5px 10px;
    font-size: 12px;
    font-weight: 800;
    color: white;
`;

const CardInformation = styled.div`
    display: flex;
    flex-direction: column;
    gap: 2px;
    margin-bottom: 5px;
    margin-top: 5px;
`;

const CardData = styled.div`
    display: flex;
    justify-content: space-between;
    gap: 12px;
    margin: 2px;
`;

const CardLabel = styled.p`
    font-size: 13px;
    font-weight: 700;
`;

const CardInfo = styled.p`
    font-size: 13px;
`;

function EmployeeCard(props) {
    let colors = ['red', 'orange', 'blue', 'purple', 'green', 'pink'];
    const color = colors[Math.floor(Math.random() * colors.length)];
    return (
        <Card key={props.user.id}>
            <CardImage color={color} src={props.user.avatar} />
            <CardTitle color={color}>{`${props.user.first_name} ${props.user.last_name}`}</CardTitle>
            <CardDesignation color={color}>{'Product Manager'}</CardDesignation>
            <CardInformation>
                <CardData>
                    <CardLabel>{"Employee Id :"}</CardLabel>
                    <CardInfo>{"123456"}</CardInfo>
                </CardData>
                <CardData>
                    <CardLabel>{"Email :"}</CardLabel>
                    <CardInfo>{props.user.email}</CardInfo>
                </CardData>
                <CardData>
                    <CardLabel>{"Mobile :"}</CardLabel>
                    <CardInfo>{"9003164637"}</CardInfo>
                </CardData>
                <CardData>
                    <CardLabel>{"Department :"}</CardLabel>
                    <CardInfo>{"Gigitech Com"}</CardInfo>
                </CardData>
            </CardInformation>
        </Card>
    )
}

export default EmployeeCard;