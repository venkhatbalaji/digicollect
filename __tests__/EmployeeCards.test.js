import React from 'react';
import EmployeeCard from '../src/component/EmployeeDetails/EmployeeCard';
import { create } from 'react-test-renderer';
import data from "./mock/employeeData";

describe('Employee Card Details unit test',()=>{
    test('testing app button', () => {
    let tree = create(<EmployeeCard user={data} />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})

