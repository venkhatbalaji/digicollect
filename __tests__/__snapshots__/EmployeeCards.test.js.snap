// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Card Details unit test testing app button 1`] = `
<div
  className="sc-bdvvtL foYwpu"
>
  <img
    className="sc-dkPtRN crUyVz"
    color="red"
  />
  <h1
    className="sc-gsDKAQ jHDxkU"
    color="red"
  >
    undefined undefined
  </h1>
  <p
    className="sc-hKwDye higtsW"
    color="red"
  >
    Product Manager
  </p>
  <div
    className="sc-eCImPb gHFksy"
  >
    <div
      className="sc-jRQBWg bMgbRf"
    >
      <p
        className="sc-gKclnd jOkCew"
      >
        Employee Id :
      </p>
      <p
        className="sc-iCfMLu iyLjXr"
      >
        123456
      </p>
    </div>
    <div
      className="sc-jRQBWg bMgbRf"
    >
      <p
        className="sc-gKclnd jOkCew"
      >
        Email :
      </p>
      <p
        className="sc-iCfMLu iyLjXr"
      />
    </div>
    <div
      className="sc-jRQBWg bMgbRf"
    >
      <p
        className="sc-gKclnd jOkCew"
      >
        Mobile :
      </p>
      <p
        className="sc-iCfMLu iyLjXr"
      >
        9003164637
      </p>
    </div>
    <div
      className="sc-jRQBWg bMgbRf"
    >
      <p
        className="sc-gKclnd jOkCew"
      >
        Department :
      </p>
      <p
        className="sc-iCfMLu iyLjXr"
      >
        Gigitech Com
      </p>
    </div>
  </div>
</div>
`;
